Mathlib/NumberTheory/NumberField/CanonicalEmbedding.lean

```diff
@@ -428,6 +431,80 @@ theorem mem_span_latticeBasis [NumberField K] (x : (E K)) :
     ext; exact mem_span_integralBasis K]
   rfl
 
+theorem mem_rat_span_latticeBasis (x : K) :
+    mixedEmbedding K x ∈ Submodule.span ℚ (Set.range (latticeBasis K)) := by
+  rw [← Basis.sum_repr (integralBasis K) x, map_sum]
+  simp_rw [map_rat_smul]
+  refine Submodule.sum_smul_mem _ _ (fun i _ ↦ Submodule.subset_span ?_)
+  rw [← latticeBasis_apply]
+  exact Set.mem_range_self i
+
+theorem latticeBasis_repr_apply (x : K) (i : ChooseBasisIndex ℤ (𝓞 K)) :
+    (latticeBasis K).repr (mixedEmbedding K x) i = (integralBasis K).repr x i := by
+  rw [← Basis.restrictScalars_repr_apply ℚ _ ⟨_, mem_rat_span_latticeBasis K x⟩, eq_ratCast,
+    Rat.cast_inj]
+  let f := (mixedEmbedding K).toRatAlgHom.toLinearMap.codRestrict _
+    (fun x ↦ mem_rat_span_latticeBasis K x)
+  suffices ((latticeBasis K).restrictScalars ℚ).repr.toLinearMap ∘ₗ f =
+    (integralBasis K).repr.toLinearMap from DFunLike.congr_fun (LinearMap.congr_fun this x) i
+  refine Basis.ext (integralBasis K) (fun i ↦ ?_)
+  have : f (integralBasis K i) = ((latticeBasis K).restrictScalars ℚ) i := by
+    apply Subtype.val_injective
+    rw [LinearMap.codRestrict_apply, AlgHom.toLinearMap_apply, Basis.restrictScalars_apply,
+      latticeBasis_apply]
+    rfl
+  simp_rw [LinearMap.coe_comp, LinearEquiv.coe_coe, Function.comp_apply, this, Basis.repr_self]
+
+variable (I : (FractionalIdeal (𝓞 K)⁰ K)ˣ)
+
+theorem det_basisOfFractionalIdeal_eq_norm
```

Comment:
Can you mention [NumberField.det_basisOfFractionalIdeal_eq_absNorm](https://leanprover-community.github.io/mathlib4_docs/Mathlib/NumberTheory/NumberField/FractionalIdeal.html#NumberField.det_basisOfFractionalIdeal_eq_absNorm) here? I mean, explaining what is the mathematical difference. Maybe even a docstring at the beginning of the section is a good idea.

Reply:
I am not sure what you have in mind. I put a docstring explaining why the result means (for me) and why it is useful. Please let me know if you're happy with that before I merge this PR. 

Thanks.